src/home_robot/home_robot/mapping/voxel.py

```diff
@@ -85,17 +231,95 @@ def write_to_pickle(self, filename: str):
             pickle.dump(data, f)
 
     def recompute_map(self):
-        """Recompute the entire map from scratch instead of doing incremental updates"""
-        self.xyz, self.feats = None, None
-        for camera_pose, xyz, feats, _ in self.observations:
+        """Recompute the entire map from scratch instead of doing incremental updates.
+        This is a helper function which recomputes everything from the beginning and migh"""
+        self.reset_cache()
+        for camera_pose, xyz, feats, base_pose, obs, _ in self.observations:
+            # TODO: logic should be the same as in "add" above
+            # TODO: switch to using just Obs struct?
             world_xyz = trimesh.transform_points(xyz, camera_pose)
             if self.xyz is None:
-                self.xyz, self.feats = world_xyz, feats
+                pc_xyz, pc_rgb = world_xyz, feats
             else:
                 pc_rgb = np.concatenate([self.feats, feats], axis=0)
                 pc_xyz = np.concatenate([self.xyz, world_xyz], axis=0)
-        pcd = numpy_to_pcd(pc_xyz, pc_rgb).voxel_down_sample(voxel_size=self.resolution)
-        self.xyz, self.feats = pcd_to_numpy(pcd)
+            if base_pose is not None:
+                self._update_visited(base_pose)
+        self._pcd = numpy_to_pcd(pc_xyz, pc_rgb).voxel_down_sample(
```

Comment:
I think this function still has the error that I was previously mentioning. Maybe it was never noticed, because this function was never used. The problem with this function is that `self.xyz` and `self.feats` are `None` after the call to the `self.reset_cache` call and they will remain `None` throughout this for loop. So at every iteration, the `if statement` will be evaluated to `True` and finally when you reach the `voxel_down_sample` line, pc_xyz and pc_rgb` will contain only the point cloud from last observation.

Reply:
I should remove this function and just repeatedly call "add"